// catalog/lib/index-standalone.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.R = exports.Radium = exports.React = exports.VideoSpecimen = exports.ReactSpecimen = exports.DownloadSpecimen = exports.TypeSpecimen = exports.ImageSpecimen = exports.HintSpecimen = exports.HtmlSpecimen = exports.ColorPaletteSpecimen = exports.ColorSpecimen = exports.CodeSpecimen = exports.AudioSpecimen = exports.mapSpecimenOption = exports.Specimen = exports.Span = exports.Page = exports.Card = exports.configureJSXRoutes = exports.configureRoutes = exports.configure = exports.render = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _Card = require('./components/Card/Card');

Object.defineProperty(exports, 'Card', {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_Card).default;
  }
});

var _Page = require('./components/Page/Page');

Object.defineProperty(exports, 'Page', {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_Page).default;
  }
});

var _Span = require('./components/Specimen/Span');

Object.defineProperty(exports, 'Span', {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_Span).default;
  }
});

var _Specimen = require('./components/Specimen/Specimen');

Object.defineProperty(exports, 'Specimen', {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_Specimen).default;
  }
});

var _mapSpecimenOption = require('./utils/mapSpecimenOption');

Object.defineProperty(exports, 'mapSpecimenOption', {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_mapSpecimenOption).default;
  }
});

var _Audio = require('./specimens/Audio');

Object.defineProperty(exports, 'AudioSpecimen', {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_Audio).default;
  }
});

var _Code = require('./specimens/Code');

Object.defineProperty(exports, 'CodeSpecimen', {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_Code).default;
  }
});

var _Color = require('./specimens/Color');

Object.defineProperty(exports, 'ColorSpecimen', {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_Color).default;
  }
});

var _ColorPalette = require('./specimens/ColorPalette');

Object.defineProperty(exports, 'ColorPaletteSpecimen', {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_ColorPalette).default;
  }
});

var _Html = require('./specimens/Html');

Object.defineProperty(exports, 'HtmlSpecimen', {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_Html).default;
  }
});

var _Hint = require('./specimens/Hint');

Object.defineProperty(exports, 'HintSpecimen', {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_Hint).default;
  }
});

var _Image = require('./specimens/Image');

Object.defineProperty(exports, 'ImageSpecimen', {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_Image).default;
  }
});

var _Type = require('./specimens/Type');

Object.defineProperty(exports, 'TypeSpecimen', {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_Type).default;
  }
});

var _Download = require('./specimens/Download');

Object.defineProperty(exports, 'DownloadSpecimen', {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_Download).default;
  }
});

var _ReactSpecimen = require('./specimens/ReactSpecimen/ReactSpecimen');

Object.defineProperty(exports, 'ReactSpecimen', {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_ReactSpecimen).default;
  }
});

var _Video = require('./specimens/Video');

Object.defineProperty(exports, 'VideoSpecimen', {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_Video).default;
  }
});

var _react = require('react');

Object.defineProperty(exports, 'React', {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_react).default;
  }
});

var _radium = require('radium');

Object.defineProperty(exports, 'Radium', {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_radium).default;
  }
});

var _ramda = require('ramda');

Object.defineProperty(exports, 'R', {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_ramda).default;
  }
});

require('babel-polyfill');

require('isomorphic-fetch');

var _specimens = require('./specimens');

var _specimens2 = _interopRequireDefault(_specimens);

var _render2 = require('./render');

var _render3 = _interopRequireDefault(_render2);

var _configure2 = require('./configure');

var _configure3 = _interopRequireDefault(_configure2);

var _configureRoutes2 = require('./configureRoutes');

var _configureRoutes3 = _interopRequireDefault(_configureRoutes2);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var mergeSpecimens = function mergeSpecimens(config, specimens) {
  return _extends({}, config, { specimens: _extends({}, specimens, config.specimens) });
};

// Configuration
var render = exports.render = function render(config, element) {
  return (0, _render3.default)(mergeSpecimens(config, _specimens2.default), element);
};
var configure = exports.configure = function configure(config) {
  return (0, _configure3.default)(mergeSpecimens(config, _specimens2.default));
};
var configureRoutes = exports.configureRoutes = function configureRoutes(config) {
  return (0, _configureRoutes3.default)(mergeSpecimens(config, _specimens2.default));
};
var configureJSXRoutes = exports.configureJSXRoutes = function configureJSXRoutes(config) {
  return (0, _configureRoutes2.configureJSXRoutes)(mergeSpecimens(config, _specimens2.default));
};

// Components
